### ISSUE TITLE
Remove hybrid Mardown+LaTeX markup from example document and add support for markdown table labels

### ISSUE THREAD

**Issue by Witiko** [danopolan/istqb_latex#75]
In the example document, we freely mix markdown with TeX commands, as enabled by [the `hybrid` option of the Markdown package][2]. This is [discouraged][3], since it makes the markdown documents less interoperable with different markdown processors. More principled alternatives exist, such as writing TeX commands in raw spans: ``` `\TeX`{=tex} ```, see also a discussion in https://github.com/Witiko/markdown/discussions/302. In ATLaS, the option `hybrid` is disabled and we may want to remove it from the example document as well.

One useful use case of TeX commands is the definition of table labels:

https://github.com/danopolan/istqb_latex/blob/cf593afa9072d92657d568c5415458b82c90dd82/example-istqb-content.md?plain=1#L54-L64

Markdown 3.0.0 is currently in alpha and [has added support for table labels][1]:

``` md
: Here is a caption describing the table. {#table:label}
```

This makes it easy for authors to reference tables in a standard way without having to use TeX commands.

In the long-time horizon, we may want to upgrade to Markdown 3.0.0 after it has had a stable release. For the time being, we may want to backport the support for table labels as [a user-defined syntax extension][4] that would be implemented in a Lua file `table-captions.lua` located in the root directory of the repository.

 [1]: https://youtu.be/U8XjTOhJkg0?t=18m52s
 [2]: https://witiko.github.io/markdown/#option-hybrid
 [3]: https://youtu.be/U8XjTOhJkg0?t=10m56s
 [4]: https://www.tug.org/TUGboat/tb43-3/tb135novotny-markdown.pdf#page=2

**Comment by Witiko** [comment]
The expected effort for removing the `hybrid` option from the example document and adding support for table captions is 4 hours. @danopolan, let me know if you'd like me to work on this.

**Comment by danopolan** [comment]
Dear @Witiko, sure, I would say that having the ability to reference tables in the document is not a priority now. So, I recommend upgrading to Markdown 3.0 when it is available to solve this.

**Comment by Witiko** [comment]
> Dear @Witiko, sure, I would say that having the ability to reference tables in the document is not a priority now. So, I recommend upgrading to Markdown 3.0 when it is available to solve this.

[Markdown 3.0.0][1] has been released last Friday. Tomorrow, I will see if we can upgrade easily or if there are issues to be solved. If yes, I will open a PR tomorrow. If not, I will document the migration steps necessary and we can postpone the upgrade until later after we have closed higher-priority tickets.

 [1]: https://github.com/Witiko/markdown/releases/tag/3.0.0
